AK/Memory.h

```diff
@@ -50,6 +52,29 @@ inline void secure_zero(void* ptr, size_t size)
     asm volatile("" ::
                      : "memory");
 }
+
+// Naive implementation of a constant time buffer comparison function.
+// The goal being to not use any conditional branching so calls are
+// guarded against potential timing attacks.
+//
+// See OpenBSD's timingsafe_memcmp  for more advanced implementations.
+inline bool timing_safe_compare(const void* b1, const void* b2, size_t len)
+{
+    auto* c1 = reinterpret_cast<const char*>(b1);
+    auto* c2 = reinterpret_cast<const char*>(b2);
```

Comment:
`static_cast` should work fine here too?

Reply:
Yup! Fixed